app/upgrades.go

```diff
@@ -361,3 +379,31 @@ func setExchangeParams(ctx sdk.Context, app *App) {
 	}
 	ctx.Logger().Info("Done ensuring exchange module params are set.")
 }
+
+// updateIbcMarkerDenomMetadata iterates markers and creates denom metadata for ibc markers
+// TODO: Remove with the saffron handlers.
+func updateIbcMarkerDenomMetadata(ctx sdk.Context, app *App) {
+	ctx.Logger().Info("Updating ibc marker denom metadata")
+	app.MarkerKeeper.IterateMarkers(ctx, func(record types.MarkerAccountI) bool {
+		if strings.HasSuffix(record.GetDenom(), "ibc/") {
+			denomTrace := transfertypes.ParseDenomTrace(record.GetDenom())
+			parts := strings.Split(denomTrace.Path, "/")
+			if len(parts) == 2 && parts[0] == "transfer" {
+				ctx.Logger().Info(fmt.Sprintf("Adding metadata to %s", record.GetDenom()))
+				chainID := app.Ics20MarkerHooks.GetChainID(ctx, parts[0], parts[1], app.IBCKeeper)
+				markerMetadata := banktypes.Metadata{
+					Base:        record.GetDenom(),
+					Name:        chainID + "/" + denomTrace.BaseDenom,
+					Display:     chainID + "/" + denomTrace.BaseDenom,
+					Description: denomTrace.BaseDenom + " from chain " + chainID,
+				}
+				err := app.MarkerKeeper.SetDenomMetaData(ctx, markerMetadata, authtypes.NewModuleAddress(types.ModuleName))
```

Comment:
I suggest just `app.BankKeeper.SetDenomMetaData` here instead of `app.MarkerKeeper.SetDenomMetaData`.
The marker one just calls the bank one then emits an event. The event is meaningless here, though. And bank one doesn't return an error either.

So using the bank one, you don't need to import either `authtypes` or `markertypes`, and you don't need the check/panic either.